packages/effect/src/internal/rateLimiter.ts

```diff
@@ -0,0 +1,74 @@
+// Direct port of https://github.com/svroonland/rezilience/blob/master/rezilience/shared/src/main/scala/nl/vroste/rezilience/RateLimiter.scala
+import type { Duration } from "effect"
+import { Chunk, Deferred, Effect, pipe, Queue, Ref, Stream } from "effect"
+
+// Find the next power of 2 greater than or equal to n.
+/** @internal */
+const nextPow2 = (n: number): number => {
+  const nextPow = Math.ceil(Math.log(n) / Math.log(2))
+  return Math.max(Math.pow(2, nextPow), 2)
+}
+
+/** @internal */
+export const make = (max: number, interval: Duration.DurationInput) => {
+  return Effect.gen(function*(_outer) {
```

Comment:
no need to use `outer / inner` JS has shadowing, you can simply use a `$`

Reply:
Done.